src/pages/ServiceList/ServiceErrorRate.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { ServiceItem } from '../../types/ServiceListComponent';
+import { Icon } from 'patternfly-react';
+
+const WARNING_THRESHOLD = 0.0;
+const ERROR_THRESHOLD = 0.2;
+
+type ServiceErrorRateProps = {
+  service: ServiceItem;
+};
+
+export default class ServiceErrorRate extends React.Component<ServiceErrorRateProps> {
+  render() {
+    return (
+      <>
+        <strong>Error rate: </strong>
+        {this.props.service.request_count > 0
```

Comment:
This (and the other clause down) should be methods, I think. They are complex enough that it becomes weird when rendering with the mixed indenting.